vendor/assets/javascripts/riot.js

```diff
@@ -1,129 +1,688 @@
+      keys = els[0].slice(1).trim().split(/,\s*/)
+    }
+
+    // clean template code
+    instance.one('mount', function() {
+      var p = dom.parentNode
+      if (p) {
+        root = p
+        root.removeChild(dom)
+      }
+    })
+
+    function startPos() {
+      return Array.prototype.indexOf.call(root.childNodes, prev) + 1
+    }
+
+    instance.on('updated', function() {
+
+      var items = tmpl(val, instance)
+          is_array = Array.isArray(items)
+
+      if (is_array) items = items.slice(0)
+
+      else {
+
+        if (!items) return // some IE8 issue
+
+        // detect Object changes
+        var testsum = JSON.stringify(items)
+        if (testsum == checksum) return
+        checksum = testsum
+
+        items = Object.keys(items).map(function(key, i) {
+          var item = {}
+          item[keys[0]] = key
+          item[keys[1]] = items[key]
+          return item
+        })
+
+      }
+
+      // remove redundant
+      diff(rendered, items).map(function(item) {
+        var pos = rendered.indexOf(item)
+        root.removeChild(root.childNodes[startPos() + pos])
+        rendered.splice(pos, 1)
+      })
+
+      // add new
+      diff(items, rendered).map(function(item, i) {
+        var pos = items.indexOf(item)
+
+        // string array
+        if (keys && !checksum) {
+          var obj = {}
+          obj[keys[0]] = item
+          obj[keys[1]] = i
+          item = obj
+        }
+
+        var tag = createTag({
+          before: root.childNodes[startPos() + pos],
+          parent: instance,
+          tmpl: template,
+          item: item,
+          root: root
+        })
+
+        instance.on('update', function() {
+          tag.update(0, true)
+        })
+
+      })
+
+      // assign rendered
+      rendered = items
+
+    })
+
+  }
+
+  riot.tag = function(name, tmpl, fn) {
+    fn = fn || noop,
+    tag_impl[name] = [tmpl, fn]
+  }
+
+  riot.mountTo = function(node, tagName, opts) {
+    var tag = tag_impl[tagName]
+    return tag && createTag({ tmpl: tag[0], fn: tag[1], root: node, opts: opts })
+  }
+
+  riot.mount = function(selector, opts) {
+    if (selector == '*') selector = Object.keys(tag_impl).join(', ')
+
+    var instances = []
+
+    each(doc.querySelectorAll(selector), function(node) {
+      if (node.riot) return
+
+      var tagName = node.tagName.toLowerCase(),
+          instance = riot.mountTo(node, tagName, opts)
+
+      if (instance) {
+        instances.push(instance)
+        node.riot = 1
+      }
+    })
+
+    return instances
+  }
+
+  // update everything
+  riot.update = function() {
+    return all_tags = all_tags.filter(function(tag) {
+      return !!tag.update()
+    })
+  }
+
+})(riot, document)
+
+
+// support CommonJS
+if (typeof exports === 'object')
+  module.exports = riot
+
+// support AMD
+else if (typeof define === 'function' && define.amd)
+  define(function() { return riot })
+
+// support browser
+else
+  this.riot = riot
 
-})({});
+})();
```

Comment:
@marcioj Can you add a line break here please ([reason](http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline)). [Heres](https://github.com/editorconfig/editorconfig/wiki/Newline-at-End-of-File-Support) how to get your editor to always do this :)